### PR TITLE
Add reset function to size Info

### PR DESCRIPTION
**Pull request by knelli2**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Review by kidder — DISMISSED** [review]

File: src/ControlSystem/ControlErrors/Size/Info.cpp [hunk 1]
```diff
@@ -42,6 +42,14 @@ void Info::pup(PUP::er& p) {
   p | discontinuous_change_has_occurred;
 }
 
+void Info::reset() {
+  suggested_time_scale = 0.0;
```
Comment (nilsvu) [inline src/ControlSystem/ControlErrors/Size/Info.cpp]: What does the value of 0 mean? Should it be NaN?

Comment (knelli2) [reply src/ControlSystem/ControlErrors/Size/Info.cpp]: 0 is a sentinel value meaning that the control system simply hasn't suggested a timescale.

Comment (nilsvu) [reply src/ControlSystem/ControlErrors/Size/Info.cpp]: Sounds like something that should be documented..

Comment (knelli2) [reply src/ControlSystem/ControlErrors/Size/Info.cpp]: Ok I documented it and added a commit to make it a better sentinel (nullopt)

**Review by nilsvu — APPROVED** [review]

**Pull request merged**